File: 029_C#/fizzbuzz_01.cs
// This is a comment.

class FizzBuzz {

    public static void cout(string s) {
        System.Console.WriteLine(s);
    }

    public static void Main() {
        for (int i = 1; i <= 100; ++i) {
            if (i % 3 == 0 & i % 5 == 0) {
                cout("FizzBuzz");
            } else if (i % 3 == 0) {
                cout("Fizz");
            } else if (i % 5 == 0) {
                cout("Buzz");
            } else {
                cout(i.ToString());
            }
        }
    }

}

File: 024_Scala/fizzbuzz_01.scala
// This is a comment.

object FizzBuzz {

    def main(args: Array[String]): Unit = {
        for (i <- 1 to 100) {
            println(fizzbuzz(i))
        }
    }

    def fizzbuzz(i: Int): String = {
        var isfizz = (i % 3 == 0)
        var isbuzz = (i % 5 == 0)
        if (isfizz && isbuzz) {
            "FizzBuzz"
        } else if (isfizz) {
            "Fizz"
        } else if (isbuzz) {
            "Buzz"
        } else {
            i.toString
        }
    }

}

